apps/speed.c

```diff
@@ -1130,6 +1166,123 @@ static int SM2_verify_loop(void *args)
 }
 #endif                         /* OPENSSL_NO_SM2 */
 
+static long kems_c[MAX_KEM_NUM][3];
+static int KEM_keygen_loop(void *args)
+{
+    loopargs_t *tempargs = *(loopargs_t **) args;
+    EVP_PKEY_CTX *ctx = tempargs->kem_gen_ctx[testnum];
+    EVP_PKEY *pkey = NULL;
+    int count;
+
+    for (count = 0; COND(kems_c[testnum][0]); count++) {
+        EVP_PKEY_keygen(ctx, &pkey);
+        /* TBD: How much does free influence runtime? */
```

Comment:
you may store and release them later 
EVP_PKEY are ref-counted,  no ?
so it will depends a lot on platform implementation

Reply:
performance tested & negligible vs randomness in keygen: Comment suitably updated in https://github.com/openssl/openssl/pull/19968/commits/b7b2a70ec374b2d556e1290a0abceb48009ae939